### DISCUSSION TITLE
Welcome to BaGet Discussions!

### DISCUSSION THREAD

**Discussion by loic-sharma** [loic-sharma/BaGet#611] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chenzuo** [comment]
希望增加私有包管理方式，包括未登录时看不到包，或者指定范围的包，并且能在包安装时验证账户密码。

**Comment by WillMarcouiller** [comment]
@loic-sharma Hi there!  I've 20+ years of experience in the software development industry.  Having an interest into having my own nuget feed, I'm interested to get started with baget.  I've been struggling for a few days now, and although I understand all the concepts of azure and cloud-readiness, I just can't get baget to deploy and run adequately in my azure container app.

Also, I'd like to bring my contribution into upgrading baget to .net6, but being unfamiliar with the project structure, I just can't figure out where should I start.  I'm not used to contribute to projects, so this would be a first experience for me.

Thx for letting me know whether I can get help from here to achieve what I wish.

**Reply by AndrewTriesToCode** [reply]
@WillMarcouiller unfortunately it seems the maintainer is busy with other items. He did do a preview release several weeks back, but otherwise not much activity. There is a recent PR, https://github.com/loic-sharma/BaGet/pull/717 that updates things to .NET 6 -- I recommend you pull that down and use it if you want to hack on the project.

**Reply by WillMarcouiller** [reply]
@AndrewTriesToCode, thx for mentioning!  I didn't know it existed already.  Also, once I pull this branch and use it, I wish to create my own business nuget feed using docker containers.  I plan on deploying to ms azure container registry and create a container app.  This works already actually.  But I don't seem to be able to upload packages to it.  When tried, I receive this error:

>  dotnet nuget push -s http://nuget-server-app.lemondesert-07557925.canadacentral.azurecontainerapps.io/v3/index.json wmi.accounting.acosdk.1.0.0.nupkg -k testapikey
Pushing wmi.accounting.acosdk.1.0.0.nupkg to 'http://nuget-server-app.lemondesert-07557925.canadacentral.azurecontainerapps.io/api/v2/package'...
  PUT http://nuget-server-app.lemondesert-07557925.canadacentral.azurecontainerapps.io/api/v2/package/
An error was encountered when fetching 'PUT http://nuget-server-app.lemondesert-07557925.canadacentral.azurecontainerapps.io/api/v2/package/'. The request will now be retried.
Error while copying content to a stream.
  Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host..
  An existing connection was forcibly closed by the remote host.
  PUT http://nuget-server-app.lemondesert-07557925.canadacentral.azurecontainerapps.io/api/v2/package/
An error was encountered when fetching 'PUT http://nuget-server-app.lemondesert-07557925.canadacentral.azurecontainerapps.io/api/v2/package/'. The request will now be retried.
Error while copying content to a stream.
  Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host..
  An existing connection was forcibly closed by the remote host.
  PUT http://nuget-server-app.lemondesert-07557925.canadacentral.azurecontainerapps.io/api/v2/package/
error: Error while copying content to a stream.
error:   Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host..
error:   An existing connection was forcibly closed by the remote host.

I wonder whether it's because the website container app is so slow (because it is, probably due to the chosen options within azure)?  And yes, the url is the right one.  So if you want to test it too, you may.  I'm just running some tests to learn and make it work for now.

**Reply by AndrewTriesToCode** [reply]
I think it might because you need to use https instead of http. You will probably get more response posting in the GitHub issues rather than in the discussions area. Which index type are you using and which storage type? Do you have the correct environment variables setup?

**Reply by WillMarcouiller** [reply]
> I think it might because you need to use https instead of http. You will probably get more response posting in the GitHub issues rather than in the discussions area. Which index type are you using and which storage type? Do you have the correct environment variables setup?

tbh, I guess I'd need some help to set this all up.  First time ever I try to deploy a cloud container app.  What's best way to get in touch with this?  I'm not accustomed to this platform, despite the years of software development experience.

**Reply by AndrewTriesToCode** [reply]
Hi again, I'm not sure about  container apps because I use app service. I was just thinking changing from http to https on the url used in the dotnet NuGet command might work for you. If you just need a private feed maybe check out feedz.io
